src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -863,10 +652,10 @@ private boolean placeBlock(@NotNull final BlockPos pos, final Block block, @NotN
         }
 
         //Workaround as long as we didn't rescan all of our buildings since BlockStairs now have different metadata values.
-        if (blockState.getBlock() instanceof BlockStairs
+        if (stateToPlace.getBlock() instanceof BlockStairs
```

Comment:
Again, these kind of if switches can easily be replaced with a System that is based on interfaces (event functional ones) that make these kind of methods better manageable then they are now.

Clearing out some of to Sonar issues in the proces.